src/components/WindowSideBarCanvasPanel.js

```diff
@@ -75,7 +76,9 @@ export class WindowSideBarCanvasPanel extends Component {
               indicatorColor="primary"
               textColor="primary"
             >
-              <Tooltip title={t('tableOfContentsList')} value="tableOfContents"><Tab className={classes.variantTab} value="tableOfContents" aria-label={t('tableOfContentsList')} aria-controls={`tab-panel-${id}`} icon={<TocIcon style={{ transform: 'scale(-1, 1)' }} />} /></Tooltip>
+              {treeStructure && treeStructure.nodes && (treeStructure.nodes.length > 0) && (
```

Comment:
Instead of doing this logic in the component, maybe the container could pass it in as a prop (`showToc`) to reduce rerenders?

Reply:
Good call.. I've made the switch.